app/src/main/java/org/mozilla/fenix/settings/PairFragment.kt

```diff
@@ -61,10 +61,9 @@ class PairFragment : Fragment(), BackHandler {
     }
 
     override fun onBackPressed(): Boolean {
-        return when {
-            qrFeature.onBackPressed() -> true
-            else -> false
-        }
+        qrFeature.onBackPressed()
+        fragmentManager?.popBackStack()
```

Comment:
Generally we shouldn't be using the FragmentManager with the Navigation Library so let's use 
```
findNavController(this@PairFragment).navigateUp()
```